doc/throttle.md

```diff
@@ -0,0 +1,158 @@
+## Overview
+
+Amoc throttle is a module that allows limiting number of users' actions per given interval,
+no matter how many users there are in a test.
+It works in both local and distributed environment, allows for dynamic rate changes during a test
+ and exposes metrics which show number of requests and executions.
+
+Amoc throttle allows to set execution `Rate` per `Interval` or limit number of parallel executions
+ when `Interval` is set to `0`. Each `Rate` is identified with a `Name`.
+Rate limiting mechanism is based on allowing execution in response to a request only when it would not exceed given `Rate`.
+Amoc throttle makes sure that given `Rate` per `Interval` is maintained on possibly equal level the whole time.
+In other words, it will not create bursts of executions which would also technically result in the same rate in a given interval.
+Because of that it may happen that the actual `Rate` will be little lower then demanded.
+It will however never be exceeded. 
+
+## API
+
+Functions provided by amoc throttle:
+
+- `start/2`, `start/3`, `start/4` - functions starting throttle mechanism for a given `Name` with a given `Rate`.
```

Comment:
It's a bit hard to read if I don't see name of the parameters in order and their types. I think when describing an API it's worth to do it a way similar to Erlang or Elixir docs where not only the arity of a function is shown but also name of the arguments and possibly their types.

Reply:
Sure, the names will make it much more readable. I could put types in but it I don't know if such edoc-like structure will look good in markdown. If you have one, could you show me an example of markdown doc formatted in such way?

Reply:
For instance this https://github.com/esl/exometer_core/blob/master/docs/exometer.md#function-details, it's basically rendering Erlang doc from the source code to markdown file. I'm not telling we need to do exactly this, but could be an inspiration :) Also @aleklisi did sth similar in his PR #102